.github/workflows/code-review.yaml

```diff
@@ -0,0 +1,51 @@
+# This workflow will perform the code review on each code change.
+# It uses "actions/cache" and turborepo in order to perform tasks only on changed files.
+
+name: "Code Review"
+
+on:
+  push:
+    branches: ["main", "master"]
+    paths:
+      - apps/**
+      - packages/**
+  pull_request:
+    types: [opened, synchronize]
+    paths:
+      - apps/**
+      - packages/**
+  workflow_dispatch: {}
+
+jobs:
+  code_review:
+    timeout-minutes: 5
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab #v3.5.2
+        with:
+          fetch-depth: 2
+
+      - name: Cache turbo build setup
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 #v3.3.1
+        with:
+          path: .turbo
+          key: ${{ runner.os }}-turbo-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-turbo-
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v3.6.0
+        with:
+          node-version: 16
```

Comment:
You could also use `node-version-file`, instead of `node-version`, specifying the `.node-version` file in the repository which, I assume, specify the Node.JS version to use.
[More informations about that here](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file).